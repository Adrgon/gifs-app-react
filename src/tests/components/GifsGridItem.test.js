import React from 'react'
import PropTypes from 'prop-types'
const { shallow } = require("enzyme")
const { GifGridItem } = require("../../components/GifGridItem")

describe('Pruebas en <GifGridItem />', ()=>{
    const title = 'Un titulo';
    const url = 'https://localhost/algo.jpg'
    const wrapper = shallow(<GifGridItem title={ title } url={ url } /> )
    
    
    test('debe mostrar el componente correctamente', ()=>{
        expect(wrapper).toMatchSnapshot()
    })

    test('debe de tener un h4 con el title', () => {
        const h4 = wrapper.find('h4');
        expect(h4.text().trim()).toBe(title)
    })

    test('debe de tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img')
        expect( img.prop('src')).toBe(url)
        expect( img.prop('alt')).toBe(title)
    })
    
    test('debe de tener animate__fadeIn', () => {
        const div = wrapper.find('div')
        const className = div.prop('className')
        expect( className.includes('animate__fadeIn')).toBe(true)
    })
        
})

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}