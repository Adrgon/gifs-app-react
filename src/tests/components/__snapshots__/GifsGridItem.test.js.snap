// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <h4>
    Un titulo
  </h4>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
</div>
`;
